presto-main/src/main/java/com/facebook/presto/operator/MultiChannelGroupByHash.java

```diff
@@ -133,6 +133,26 @@ public MultiChannelGroupByHash(
     }
 
     @Override
+    public int compare(int leftGroupId, int rightGroupId)
+    {
+        long leftAddress = groupAddressByGroupId.get(leftGroupId);
+        int leftBlockIndex = decodeSliceIndex(leftAddress);
+        int leftPosition = decodePosition(leftAddress);
+        long rightAdrress = groupAddressByGroupId.get(rightGroupId);
+        int rightBlockIndex = decodeSliceIndex(rightAdrress);
+        int rightPosition = decodePosition(rightAdrress);
+
+        for (int column = 0; column < channels.length; column++) {
+            int compare = types.get(column).compareTo(channelBuilders.get(column).get(leftBlockIndex), leftPosition,
```

Comment:
This is going to throw an exception if the type isn't orderable. I think you should add a check in the planner/analyzer that fails the query with a nice error message saying that spillable aggregations only work with orderable types